### PR TITLE
Develop mobile app in rails 

### PR DESCRIPTION
**Pull request by KanoCode**
In this project the following was done:

- Built  the application using models from this schema diagram
erd_diagram

Implemented the authentication and authorization with devise and cancancan gems

**The projects has the following features**
     - Created the page to add a category.
     - Created the home page.
      -  Created the page to add a transaction.
      - Created the transactions page.
  
  **Wrote unit tests and integration tests for:**
    - Category model
    - User model
    - transaction model

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :categories
+  has_many :payments
+
+  validates :name, :email, :password, presence: true
```
Comment (OLIPLICHE) [inline app/models/user.rb:10]: ## Nice Job in using device for authentication 🥇

File: app/views/transaction/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+      <%= form.select :category_id, Category.order(:name).where(user_id:current_user.id).pluck(:name, :id), {}, {:multiple => true, required:true, class: "form-select mb-3"} %>
+  <div>
+    <%= form.label :amount, class:"form-label" %><br>
+    <%= form.number_field :amount, class:"form-control" %>
+  </div>   
+
+  <div class="actions login text-center py-1 my-2">
+    <%= form.submit "Add Transaction",class:"btn text-light " %>
+  </div>
```
Comment (OLIPLICHE) [inline app/views/transaction/new.html.erb:27]: ## Congratulations:
Capture

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  describe 'associations' do
+    it 'has many payments' do
+      assc = described_class.reflect_on_association(:transactions)
+      expect(assc.macro).to eq :has_many
+    end
+    it 'belongs to user' do
+      assc = described_class.reflect_on_association(:user)
+      expect(assc.macro).to eq :belongs_to
```
Comment (OLIPLICHE) [inline spec/models/category_spec.rb:26]: ## Units test all passed: 
test1
test2
test3

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (OLIPLICHE) [inline spec/models/user_spec.rb:3]: ## Integration test Passed 🥇 

Capture

**Review by OLIPLICHE — APPROVED** [review]
Hi @KanoCode, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- The App is working fine 👏 👏
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏

## Optional: Take note of this
- Reading from Requirements, YOUR WERE SUPPOSED TO USE GITFLOW AND NOT GITHUB FLOW
- Note: I did not invalidate your project because When you change the base branch of your pull request, some commits may be removed from the timeline. Review comments may also become outdated, as the line of code that the comment referenced may no longer be part of the changes in the pull request. 
- Next time, Your Work will be marked Invalid
- Learn more: [Gitflow](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) vs [Github flow](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request)
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 4 above)
Comment (KanoCode) [reply spec/models/user_spec.rb:3]: thank you @OLIPLICHE for the constructive feedback

**Pull request merged**